src/Compilers/CSharp/Portable/Symbols/Source/SourceComplexParameterSymbol.cs

```diff
@@ -262,6 +262,36 @@ private ConstantValue DefaultSyntaxValue
                         }
                     }
 
+                    // Ensure availability of `DecimalConstantAttribute`.
+                    // For lambdas, check also explicit attribute application
+                    // because it will be re-synthesized in the delegate type.
+                    var isLambda = this is LambdaParameterSymbol;
+                    var defaultValueLocation = parameterEqualsValue?.Value.Syntax.Location ?? CSharpSyntaxNode?.Location ?? Locations.FirstOrDefault();
+                    if (isLambda
+                        ? _lazyDefaultSyntaxValue is { IsDecimal: true } ^
+                            (HasOptionalAttribute && DefaultValueFromAttributes is { IsDecimal: true })
```

Comment:
>IsDecimal: true [](http://example.com/codeflow?start=85&length=15)

This check doesn't feel necessary because we require default values from attributes and from values to match and that check is performed elsewhere.

Reply:
Thanks, you are right. Removed.